### DISCUSSION TITLE
image processing methods

### DISCUSSION THREAD

**Discussion by v923z** [v923z/micropython-ulab#667] [Ideas]
Taking ideas from https://github.com/v923z/micropython-ulab/issues/363, this discussion should explore, what kind of image processing methods could be useful in the context of a microcontroller. A good first candidate could be openmv's [projective transforms](https://github.com/AprilRobotics/apriltag/blob/master/common/homography.c), as well as blob searching. 

@kwagyeman
@teuler

**Comment by kwagyeman** [comment]
@v923z - OpenMV maintains a library of image-processing functions. I don't have any interest in methods outside of this. I can implement and export homogrphy support into our code base directly. However, I thought it would make more sense to have it in ulab. That said, homography is a computer vision like function... so, maybe it doesn't belong in ulab anyway? It's matrix math, though.

If svd support is exported in ulab then I can add it to openmv firmware directly.

**Reply by v923z** [reply]
When I started out with this project, I didn't name it `unumpy` or anything like that, because I thought that there could be functionality that is not in `numpy` or `scipy`, but might, in a way, fit into the scope of what I envisioned. And such functions are exactly the ones that I had in mind.

So, I wouldn't mind adapting functions that might have traction in the community, for the simple reason that by moving themppppppppppop would make them accessible to a bigger audience. On the other hand, I don't want to re-implementpppi@ everything here just for kicks. If no-one is interested in blobl searching, then there is no point in having a function for that.
Op
SVD was requested here in a different context, so I think it was the right decision to implement the function here.ll

**Reply by kwagyeman** [reply]
Once you start adding vision functions, people will want support, and then you to make them faster. :)

I think the homography stuff is useful, though, for non-vision functions, too. It gives you the matrix you need to map a plan to 4 points.
